src/resdk/query.py

```diff
@@ -184,8 +184,25 @@ def _dehydrate_resources(self, obj):
     def _add_filter(self, filter_: dict):
         """Add filtering parameters."""
         for key, value in filter_.items():
-            # 'sample' is called 'entity' in the backend.
-            key = key.replace("sample", "entity")
+            # Make best-effort to rename fields to server fields.
+            resource = self.resource
```

Comment:
Any reason to assign `self.resource` to a new variable? IMO this is unneccessary.

Reply:
See line 202 below.